### ISSUE TITLE
Possible D3D12 corruption

### ISSUE THREAD

**Issue by rorydriscoll** [baldurk/renderdoc#1806]
## Description

I ran a capture of my engine for frame 0 to try to determine why my IBL filtering appears to be broken. During frame 0, I upload a bunch of textures and meshes to the GPU using a copy queue, then render the first frame as normal. When I look at these textures in RenderDoc, they appear to be mixed up between the expected texture and the texture that is uploaded immediately afterwards. Also, the meshes appear to have holes in them.

Note that if I capture a later frame then things all appear to be good. I'm waiting on a fence after each upload, and also transitioning the resources to the proper state on the graphics queue before attempting to read them.

I assumed that this is a bug in my upload code, but I notice that the textures change each time I look at them. I also notice that the thumbnail for the frame looks correct, but when looking at the final frame buffer in RenderDoc it looks wrong, so I'm not so sure anymore.

## Steps to reproduce

Capture is [here](https://www.dropbox.com/s/50d2ywf4t0a8eeo/Corruption.rdc?dl=0).

## Environment

* RenderDoc version: 1.7
* Operating System: Windows 10 1909
* Graphics API: D3D12

Running on GeForce RTX 2080 Ti, driver is 442.74

**Comment by rorydriscoll** [comment]
I was able to look at the first frame in Pix, and textures/meshes look correct there so I think that something probably is off in the RenderDoc replay.

**Comment by baldurk** [comment]
That commit should fix the issue - I was replaying Signal/Wait synchronisation on the wrong queue so all those copy uploads were executing while I was replaying mapped resource writes on the CPU. I never saw the texture corruption for some reason (possibly due to running a debug build which was slower) but I did catch mesh corruption.

btw I did notice one other minor strange thing in the capture while investigating this which I couldn't explain. Would it be possible for you to share your application with me so I can debug the capture-time work? Either here or privately.
